server/tserver/src/main/java/org/apache/accumulo/tserver/ScanServer.java

```diff
@@ -572,12 +660,18 @@ public MultiScanResult continueMultiScan(TInfo tinfo, long scanID, long busyTime
       throws NoSuchScanIDException, TSampleNotPresentException, TException {
     LOG.debug("continue multi scan: {}", scanID);
 
-    return handler.continueMultiScan(tinfo, scanID, busyTimeout);
+    try {
+      return handler.continueMultiScan(tinfo, scanID, busyTimeout);
+    } catch (Exception e) {
+      deleteScanReferences(scanID, true);
+      throw e;
+    }
   }
 
   @Override
   public void closeMultiScan(TInfo tinfo, long scanID) throws NoSuchScanIDException, TException {
     LOG.debug("close multi scan: {}", scanID);
+    deleteScanReferences(scanID, true);
```

Comment:
A client may never call this RPC.  Also its possible a client could call it more than once.  Either one of these could throw off the reference counting.